lib/atrium/member.rb

```diff
@@ -1,20 +1,141 @@
+      accounts_response = ::Atrium.client.make_request(:get, endpoint)
+
+      account = accounts_response["accounts"].map do |account|
+        ::Atrium::Account.new(account)
+      end
+    end
+
+    def aggregate
+      endpoint = "users/#{self.user_guid}/members/#{self.guid}/aggregate"
+      member_response = ::Atrium.client.make_request(:post, endpoint)
+
+      member_params = member_response["member"]
+      self.assign_attributes(member_params)
+      self
+    end
+
+    def aggregation_status
+      endpoint = "/users/#{self.user_guid}/members/#{self.guid}/status"
+      member_response = ::Atrium.client.make_request(:get, endpoint)
+
+      member_params = member_response["member"]
+      self.assign_attributes(member_params)
+      self
+    end
+
+    def challenges
+      endpoint = "/users/#{self.user_guid}/members/#{self.guid}/challenges"
+      member_response = ::Atrium.client.make_request(:get, endpoint)
+
+      member_params = member_response["member"]
+      self.assign_attributes(member_params)
+      self
+    end
+
+    def delete
+      endpoint = "/users/#{self.user_guid}/members/#{self.guid}"
+      ::Atrium.client.make_request(:delete, endpoint)
+
+      self
+    end
+
+    def read_account(account_guid:)
+      endpoint = "/users/#{self.user_guid}/members/#{self.guid}/accounts/#{account_guid}"
+      account_response = ::Atrium.client.make_request(:get, endpoint)
+
+      account_params = account_response["account"]
+      ::Atrium::Account.new(account_params)
+    end
+
+    def resume(challenge_credentials)
+      endpoint = "/users/#{self.user_guid}/members/#{self.guid}/resume"
+      body = resume_params(challenge_credentials)
+      member_response = ::Atrium.client.make_request(:get, endpoint, body)
+
+      member_params = member_response["member"]
+      self.assign_attributes(member_params)
+      self
+    end
+
+    def update(params)
+      endpoint = "/users/#{self.user_guid}/members/#{self.guid}"
       body = member_body(params)
-      ::Atrium.client.make_request(:post, endpoint, body)
+      member_response = ::Atrium.client.make_request(:put, endpoint, body)
+
+      member_params = member_response["member"]
+      self.assign_attributes(member_params)
+      self
+    end
+
+    def transactions
+      endpoint = "users/#{self.user_guid}/members/#{self.guid}/transactions"
+      transactions_response = ::Atrium.client.make_request(:post, endpoint)
+
+      transactions = transactions_response["transactions"].map do |transaction|
+        ::Atrium::Transaction.new(transaction)
+      end
     end
 
+    private
     ##
-    # GET /users/:user_guid/members/:member_guid/status
+    # PRIVATE CLASS METHODS
     #
-    def self.status(user_guid, member_guid)
-      endpoint = "/users/#{user_guid}/members/#{member_guid}/status"
-      ::Atrium.client.make_request(:get, endpoint)
+    def self.create_params(institution_code, credentials_array)
```

Comment:
niiice!! I wasn't aware of the `private_class_method`--never had to use it before ⚡️